Travis periodic job fails on Flake8

https://travis-ci.org/pytrip/pytrip/jobs/291759726

new flake8 package was released on 23.10.2017, probably with stricter rules
https://pypi.python.org/pypi/flake8


```
/home/travis/.travis/job_stages: line 169: 5578 Terminated travis_jigger $! $timeout $cmd
GLOB sdist-make: /home/travis/build/pytrip/pytrip/setup.py
packaging installed: check-manifest==0.35,cycler==0.10.0,flake8==3.5.0,matplotlib==1.5.3,mccabe==0.6.1,numpy==1.13.3,pycodestyle==2.3.1,pydicom==0.9.9,pyflakes==1.6.0,pyparsing==2.2.0,python-dateutil==2.6.1,pytz==2017.2,six==1.11.0
packaging runtests: PYTHONHASHSEED='2628226306'
packaging runtests: commands[0] | check-manifest
lists of files in version control and sdist match
packaging runtests: commands[1] | flake8 .
./pytrip/vdx.py:46:1: E722 do not use bare except'
./pytrip/paths.py:57:9: E741 ambiguous variable name 'l'
./pytrip/paths.py:58:9: E741 ambiguous variable name 'l'
./pytrip/paths.py:62:9: E741 ambiguous variable name 'l'
./pytrip/dos.py:38:1: E722 do not use bare except'
./pytrip/cube.py:36:1: E722 do not use bare except'
./pytrip/tripexecuter/execute.py:39:1: E722 do not use bare except'
./pytrip/tripexecuter/execute.py:503:17: E722 do not use bare except'
./pytrip/tripexecuter/execute.py:512:17: E722 do not use bare except'
./pytrip/tripexecuter/execute.py:536:17: E722 do not use bare except'
./pytrip/tripexecuter/execute.py:545:17: E722 do not use bare except'
./tests/test_models.py:86:9: E741 ambiguous variable name 'l'
./tests/test_let.py:42:9: E741 ambiguous variable name 'l'
./examples/example00_basic.py:43:1: E741 ambiguous variable name 'l'
ERROR: InvocationError: '/home/travis/build/pytrip/pytrip/.tox/packaging/bin/flake8 .'
```

See also: https://github.com/DataMedSci/pymchelper/issues/269

--- a/examples/example00_basic.py
+++ b/examples/example00_basic.py
@@ -40,5 +40,5 @@ l2 = pt.LETCube()
 l1.read("box052000.dosemlet.dos")
 l2.read("box053000.dosemlet.dos")
 
-l = ((d1 * l1) + (d2 * l2)) / (d1 + d2)
-l.write("out.dosemlet.dos")
+let = ((d1 * l1) + (d2 * l2)) / (d1 + d2)
+let.write("out.dosemlet.dos")

--- a/pytrip/cube.py
+++ b/pytrip/cube.py
@@ -33,7 +33,7 @@ try:
     from dicom import UID
 
     _dicom_loaded = True
-except:
+except ImportError:
     _dicom_loaded = False
 
 from pytrip.error import InputError, ModuleNotLoadedError, FileNotFound

--- a/pytrip/dos.py
+++ b/pytrip/dos.py
@@ -35,7 +35,7 @@ try:
     from dicom import UID
 
     _dicom_loaded = True
-except:
+except ImportError:
     _dicom_loaded = False
 
 logger = logging.getLogger(__name__)

--- a/pytrip/paths.py
+++ b/pytrip/paths.py
@@ -54,13 +54,14 @@ class DensityProjections:
     def calculate_quality_grid(self, voi, gantry, couch, calculate_from=0, stepsize=1.0, avoid=[], gradient=True):
         """ TODO: Documentation
         """
-        l = self.calculate_quality_list(voi, gantry, couch, calculate_from, stepsize, avoid=avoid, gradient=gradient)
-        l = sorted(l, key=cmp_to_key(cmp_sort))
+        result = self.calculate_quality_list(voi, gantry, couch, calculate_from, stepsize,
+                                             avoid=avoid, gradient=gradient)
+        result = sorted(result, key=cmp_to_key(cmp_sort))
         grid_data = []
-        for x in l:
+        for x in result:
             grid_data.append(x["data"][0])
-        l = np.reshape(grid_data, (len(gantry), len(couch)))
-        return l
+        result = np.reshape(grid_data, (len(gantry), len(couch)))
+        return result
 
     def calculate_quality_list(self, voi, gantry, couch, calculate_from=0, stepsize=1.0, avoid=[], gradient=True):
         """ TODO: Documentation

--- a/pytrip/tripexecuter/execute.py
+++ b/pytrip/tripexecuter/execute.py
@@ -36,7 +36,7 @@ logger = logging.getLogger(__name__)
 
 try:
     import paramiko
-except:
+except ImportError:
     logger.warning("Paramiko package not installed, only local TRiP access possible")
 
 
@@ -500,7 +500,7 @@ class Execute(object):
                 # login with provided username + empty password
                 try:
                     ssh.connect(self.servername, username=self.username, password="")
-                except:
+                except Exception:
                     logger.error("Cannot connect to " + self.servername)
                     logger.error("Check username, password or key in " + self.rsakey_local_path)
                     raise
@@ -509,7 +509,7 @@ class Execute(object):
                 rsa_key = paramiko.RSAKey.from_private_key_file(rsa_keypath)
                 try:
                     ssh.connect(self.servername, username=self.username, pkey=rsa_key)
-                except:
+                except Exception:
                     logger.error("Cannot connect to " + self.servername)
                     logger.error("Check username and your key in " + self.rsakey_local_path)
                     raise
@@ -533,7 +533,7 @@ class Execute(object):
                 # login with provided username + empty password
                 try:
                     transport.connect(username=self.username, password="")
-                except:
+                except Exception:
                     logger.error("Cannot connect to " + self.servername)
                     logger.error("Check username, password or key in " + self.rsakey_local_path)
                     raise
@@ -542,7 +542,7 @@ class Execute(object):
                 rsa_key = paramiko.RSAKey.from_private_key_file(rsa_keypath)
                 try:
                     transport.connect(username=self.username, pkey=rsa_key)
-                except:
+                except Exception:
                     logger.error("Cannot connect to " + self.servername)
                     logger.error("Check username and your key in " + self.rsakey_local_path)
                     raise

--- a/pytrip/vdx.py
+++ b/pytrip/vdx.py
@@ -43,7 +43,7 @@ try:
     from dicom import UID
 
     _dicom_loaded = True
-except:
+except ImportError:
     _dicom_loaded = False
 
 logger = logging.getLogger(__name__)

--- a/tests/test_let.py
+++ b/tests/test_let.py
@@ -39,14 +39,14 @@ class TestLet(unittest.TestCase):
         self.vdx000 = os.path.join(testdir, "tst003000.vdx")
 
     def test_read(self):
-        l = LETCube()
-        l.read(self.let001)
+        let = LETCube()
+        let.read(self.let001)
 
-        v = create_sphere(l, name="sph", center=[10, 10, 10], radius=8)
+        v = create_sphere(let, name="sph", center=[10, 10, 10], radius=8)
         self.assertIsNotNone(v)
 
         logger.info("Calculating DVH")
-        result = l.calculate_lvh(v)
+        result = let.calculate_lvh(v)
         self.assertIsNotNone(result)
         lvh, min_l, max_l, mean, area = result
         self.assertGreater(area, 2.0)
@@ -55,12 +55,12 @@ class TestLet(unittest.TestCase):
         self.assertEqual(min_l, 0.0)
         self.assertEqual(max_l, 1.0)
 
-        self.assertGreater(l.get_max(), 30.0)
+        self.assertGreater(let.get_max(), 30.0)
 
         fd, outfile = tempfile.mkstemp()
         os.close(fd)  # Windows needs it
         os.remove(outfile)  # we need only temp filename, not the file
-        l.write(outfile)
+        let.write(outfile)
         hed_file = outfile + LETCube.header_file_extension
         dos_file = outfile + LETCube.data_file_extension
         self.assertTrue(os.path.exists(hed_file))

--- a/tests/test_models.py
+++ b/tests/test_models.py
@@ -81,18 +81,18 @@ class TestProton(unittest.TestCase):
     def test_mcnamara_cube(self):
         """ McNamara test on real cubes.
         """
-        d = pt.DosCube()
-        d.read(self.cube001)
-        l = pt.LETCube()
-        l.read(self.cube001)
-        v = pt.VdxCube(d)
+        dose = pt.DosCube()
+        dose.read(self.cube001)
+        let = pt.LETCube()
+        let.read(self.cube001)
+        v = pt.VdxCube(dose)
         logger.info("Adding VDX from path " + self.vdx)
         v.read(self.vdx)
 
         # increasing LET should increase RBE
         abx = 10.0  # alpha/beta ratio for x-rays [Gy]
-        rbe1 = rbe_mcnamara(d.cube, l.cube, abx)
-        rbe2 = rbe_mcnamara(d.cube, l.cube, 2.0)
+        rbe1 = rbe_mcnamara(dose.cube, let.cube, abx)
+        rbe2 = rbe_mcnamara(dose.cube, let.cube, 2.0)
 
         self.assertTrue(np.all(rbe2 >= rbe1))  # RBE goes up as abx goes down.
 

